### DISCUSSION TITLE
A real Ai Upscaler/Enlarger (CPU based)

### DISCUSSION THREAD

**Discussion by Daijobou** [ImageMagick/ImageMagick#6677] [Contribute]
### Is your feature request related to a problem? Please describe.

Enlarge images (like 2 times) is with ImageMagick blurred and in most cases (for me) not really usable. Its unfortunately true for most graphics tools. GIMP is also a disaster here. ImageMagick is a great tool, I love it, however shouldn't it also be able to enlarge images in a modern way?

### Describe the solution you'd like

There exists tools since many years that do this work excellent. The most avatars in communities have cartoon-look, so here open source ai upscaler specialized for cartoons as example:
https://github.com/nagadomi/waifu2x
https://github.com/bloc97/Anime4K 

Unfortunately this sources are poorly maintained or stalled. I would like, that ImageMagick add a CPU based (for webserver) AI Upscaler as new function.  For me, it's not a problem that this function is much slower.

However I understand when this is not what ImageMagick want to offer in this lib. I just wanted to throw it in, because for webserver there exists not such a lib (not that I know it) and ImageMagick would be here a **_must have_** for anyone looking for something like that. ;)

**Comment by fmw42** [comment]
See graphics upscales at https://imagemagick.org/script/command-line-options.php#magnify

**Comment by Daijobou** [comment]
Thanks for this hint. I tested it with PHP magnifyImage() and is not what I expect as a result.

Original
![image](https://user-images.githubusercontent.com/260343/187971057-f089bb7d-f107-4cec-b114-210b90aa0282.png)

Results; Left side converted with [waifu2x](http://waifu2x.udp.jp/index.html), right side its ImageMagick with [magnifyImage()](https://www.php.net/manual/en/imagick.magnifyimage.php)
![image](https://user-images.githubusercontent.com/260343/187971256-b10698d9-74eb-4df0-a25e-179957a3bb63.png)

Waifu2x needed 1.2s for converting.

**Comment by PeterBBBBB** [comment]
There is a command line tool, xbrzscale, that is easier to build and use than Waifu2x.
Does scaling up to x6 in one pass, similar quality to Waifu2x
